File: app/src/main/java/com/vuiya/bookbuddy/LibraryAdapter.java
package com.vuiya.bookbuddy;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class LibraryAdapter extends RecyclerView.Adapter<LibraryAdapter.LibraryViewHolder> {

    private List<LibraryItem> libraryItems;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(LibraryItem item);
        void onDeleteClick(LibraryItem item);
    }

    public LibraryAdapter(List<LibraryItem> libraryItems, OnItemClickListener listener) {
        this.libraryItems = libraryItems;
        this.listener = listener;
    }

    @NonNull
    @Override
    public LibraryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_library, parent, false);
        return new LibraryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LibraryViewHolder holder, int position) {
        LibraryItem item = libraryItems.get(position);
        holder.bind(item);
    }

    @Override
    public int getItemCount() {
        return libraryItems.size();
    }

    class LibraryViewHolder extends RecyclerView.ViewHolder {
        private TextView tvTitle, tvAuthor, tvLanguage, tvType;
        private ImageButton btnDelete;

        public LibraryViewHolder(@NonNull View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.tv_title);
            tvAuthor = itemView.findViewById(R.id.tv_author);
            tvLanguage = itemView.findViewById(R.id.tv_language);
            tvType = itemView.findViewById(R.id.tv_type);
            btnDelete = itemView.findViewById(R.id.btn_delete);
        }

        public void bind(LibraryItem item) {
            tvTitle.setText(item.getTitle());
            tvAuthor.setText(item.getAuthor());
            tvLanguage.setText(item.getLanguage());
            tvType.setText(item.getType());

            // Set click listeners
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClick(libraryItems.get(position));
                        }
                    }
                }
            });

            btnDelete.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onDeleteClick(libraryItems.get(position));
                        }
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/vuiya/bookbuddy/LibraryAdapter.kt
package com.vuiya.bookbuddy

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class LibraryAdapter(
    private val libraryItems: List<LibraryItem>,
    private val listener: OnItemClickListener
) : RecyclerView.Adapter<LibraryAdapter.LibraryViewHolder>() {

    interface OnItemClickListener {
        fun onItemClick(item: LibraryItem)
        fun onDeleteClick(item: LibraryItem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LibraryViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_library, parent, false)
        return LibraryViewHolder(view)
    }

    override fun onBindViewHolder(holder: LibraryViewHolder, position: Int) {
        val item = libraryItems[position]
        holder.bind(item)
    }

    override fun getItemCount(): Int = libraryItems.size

    inner class LibraryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvTitle: TextView = itemView.findViewById(R.id.tv_title)
        private val tvAuthor: TextView = itemView.findViewById(R.id.tv_author)
        private val tvLanguage: TextView = itemView.findViewById(R.id.tv_language)
        private val tvType: TextView = itemView.findViewById(R.id.tv_type)
        private val btnDelete: ImageButton = itemView.findViewById(R.id.btn_delete)

        fun bind(item: LibraryItem) {
            tvTitle.text = item.title
            tvAuthor.text = item.author
            tvLanguage.text = item.language
            tvType.text = item.category

            itemView.setOnClickListener {
                val position = adapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    listener.onItemClick(libraryItems[position])
                }
            }

            btnDelete.setOnClickListener {
                val position = adapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    listener.onDeleteClick(libraryItems[position])
                }
            }
        }
    }
}
